resolver-dns-native-windows/src/main/c/src/JavaInterface.cpp

```diff
@@ -0,0 +1,63 @@
+#include "JavaInterface.h"
+#include "NetworkAdapters.h"
+#include "NativeExceptionClass.h"
+#include <memory>
+#include "JavaNetworkAdapterClass.h"
+#include <stdexcept>
+
+#define NETTY_JNI_UTIL_JNI_VERSION JNI_VERSION_1_6
```

Comment:
Can we just use c ? We dont use any c++ atm and I would like to keep it like this to make things easier for the maintainers .

Reply:
Ported the native code to c.